pkg/devicetemplates/ttscsv.go

```diff
@@ -33,7 +33,24 @@ import (
 // TTSCSV is the device template converter ID.
 const TTSCSV = "the-things-stack-csv"
 
-type ttsCSV struct{}
+type ttsCSV struct {
+	profile EndDeviceProfileFetcher
+}
+
+// EndDeviceProfileFetcher specifies the necessary methods to ascertains if the end device contains the necessary
+// information to make the profile template request and for performing the request itself.
+type EndDeviceProfileFetcher interface {
+	ShouldFetchProfile(*ttnpb.EndDevice) bool
+	FetchProfile(context.Context, *ttnpb.EndDevice) (*ttnpb.EndDeviceTemplate, error)
+}
+
+// NewCSV allows the overwrite of the csv converter with an option interface that validates if it should fetch the End
```

Comment:
I think this can be simpler; `NewCSV adds EndDeviceProfileFetcher to the CSV converter`.

Reply:
Tried to make it simpler but I wasn't sure if the word `adds` fits the description, tried to word in a different way.